### PR TITLE
Added throughput samplers

### PR DESCRIPTION
**Pull request by vikramraman**

### CODE REVIEW

File: src/main/java/com/wavefront/sdk/common/Clock.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.wavefront.sdk.common;
+
+public abstract class Clock {
```
Comment (srujann) [inline src/main/java/com/wavefront/sdk/common/Clock.java]: java doc for the class

File: src/main/java/com/wavefront/sdk/common/RateLimiter.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package com.wavefront.sdk.common;
+
+/**
+ * A rate-limiter based on the token bucket algorithm.
+ *
+ * @author Vikram Raman
```
Comment (srujann) [inline src/main/java/com/wavefront/sdk/common/RateLimiter.java]: nit: use author format consistent with existing files.

File: src/main/java/com/wavefront/sdk/common/Clock.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    static long currTime = 0;
```
Comment (srujann) [inline src/main/java/com/wavefront/sdk/common/Clock.java]: is this only for the test? (since I see it being used only in tests)

Comment (vikramraman) [reply src/main/java/com/wavefront/sdk/common/Clock.java]: Yes, this is only for tests. For real, we'll just delegate to `System.nanoTime()`.

**Comment by panghy** [comment]
@vikramraman please address the comments or close

**Comment by vikramraman** [comment]
Closing this for now.

**Pull request closed without merging**